makePoplogCommander.sh

```diff
@@ -102,34 +157,140 @@ static Chain bump( Chain r, int n ) {
     return r;
 }
 
-Chain chain_push( Chain r, Ref ch ) {
-    bump( r, 1 );
+void vector_push( Vector r, Ref ch ) {
+    vector_bump( r, 1 );
     r->data[ r->used ] = ch;
     r->used += 1;
-    return r;
 }
 
-Chain chain_new() {
+Vector vector_new() {
     // calloc implicitly zeros size & used & sets data to NULL.
-    return (Chain)calloc( sizeof( struct Chain ), 1 );
+    return (Vector)calloc( sizeof( struct Vector ), 1 );
 }
 
-void chain_free( Chain r ) {
-    free( r->data );
-    free( r );
+void vector_free( Vector v ) {
+    free( v->data );
+    free( v );
 }
 
-int chain_length( Chain r ) {
-    return r->used;
+int vector_length( Vector v ) {
+    return v->used;
 }
 
-Ref chain_index( Chain r, int n ) {
+Ref vector_get( Vector r, int n ) {
     if ( !( 0 <= n && n < r->used ) ) {
-        mishap( "Chain index (%d) out of range (0-%d)", n, r->used );
+        mishap( "Vector index (%d) out of range (0-%d)", n, r->used );
     }
     return r->data[ n ];
 }
 
+Ref vector_set( Vector r, int n, Ref x ) {
+    if ( !( 0 <= n && n < r->used ) ) {
+        mishap( "Vector index (%d) out of range (0-%d)", n, r->used );
+    }
+    return r->data[ n ] = x;
+}
+
+Ref * vector_nc_nt_array( Vector v ) {
```

Comment:
nc nt?

Reply:
Oh whooooopsie! I meant to rename that. (nc = non-copying, nt = null-terminated)

Reply:
I have renamed `_nc_nt_` to `_as_`. I reserve the use of `as` for views.